third_party/houdini/lib/gusd/GU_PackedUSD.cpp

```diff
@@ -638,38 +642,56 @@ GusdGU_PackedUSD::save(UT_Options &options, const GA_SaveMap &map) const
 bool
 GusdGU_PackedUSD::getBounds(UT_BoundingBox &box) const
 {
+    // Box caching is handled in getBoundsCached()
+#if SYS_VERSION_FULL_INT < 0x12000000
     if( m_boundsCache.isValid() )
     {
         box = m_boundsCache;
         return true;
     }
+#endif
 
     UsdPrim prim = getUsdPrim();
 
     if( !prim ) {
-        cerr << "Invalid prim " << m_primPath << endl;
+        UT_ASSERT_MSG(0, "Invalid USD prim");
     }
 
     if(UsdGeomImageable visPrim = UsdGeomImageable(prim))
     {
         TfTokenVector purposes = GusdPurposeSetToTokens(m_purposes);
 
-        if( GusdBoundsCache::GetInstance().ComputeUntransformedBound( 
-                        prim, 
-                        UsdTimeCode( m_frame ), 
+#if SYS_VERSION_FULL_INT >= 0x12000000
+        if ( GusdBoundsCache::GetInstance().ComputeUntransformedBound(
+                prim,
+                UsdTimeCode( m_frame ),
+                purposes,
+                box )) {
+            return true;
+        }
+#else
+        if( GusdBoundsCache::GetInstance().ComputeUntransformedBound(
+                        prim,
+                        UsdTimeCode( m_frame ),
                     purposes,
                         m_boundsCache )) {
                 box = m_boundsCache;
```

Comment:
This patch could be a lot shorter if in both cases you put the result into box, then for the older version you copied box to m_boundscache. Ie:
```
    if (GuseBoundsCache::Get...(..., box)) {
    #if SYS_VERSION_FULL_INT < 0x12000000
      m_boundsCache = box;
    # endif
      return true;
   }

```

Reply:
Agreed, thank you for the suggestion!